drivers/serial/uart_nrfx_uart.c

```diff
@@ -597,14 +597,15 @@ static void rx_isr(const struct device *dev)
 		if (uart0_cb.rx_timeout == 0) {
 			rx_rdy_evt(dev);
 		} else if (uart0_cb.rx_timeout != SYS_FOREVER_MS) {
-			k_delayed_work_submit(&uart0_cb.rx_timeout_work,
-					      K_MSEC(uart0_cb.rx_timeout));
+			k_timer_start(&uart0_cb.rx_timeout_timer,
+				      K_MSEC(uart0_cb.rx_timeout),
+				      K_MSEC(uart0_cb.rx_timeout));
```

Comment:
Why  are these repeating rather than one-shot (third argument `K_NO_WAIT`)?  Doesn't this cause the handler to be invoked repeatedly even when there's no data?

Reply:
My bad. I blindly copied that from `uart_nrfx_uarte.c` where this timer is used in a slightly different way (there it is supposed to be invoked repeatedly). Thanks for catching this!

Reply:
Fixed. I extended also a bit the `test_single_read` test case so that mistakes like this could be caught there.